### PR TITLE
Add saptune solution apply to formula

### PR DESCRIPTION
**Pull request by MalloZup**
implement latest module enablement to the formula via pillar

### CODE REVIEW

**Comment by MalloZup** [comment]
I will test this right now but imho it could work

File: pillar.example [hunk 1]
```diff
@@ -8,6 +8,8 @@ hana:
       sid: 'prd'
       instance: 00
       password: 'Qwerty1234'
+      # saptune solution to apply ( by default nothing is applied)
+      saptune_solution: 'MAXDB'
```
Comment (arbulu89) [inline pillar.example]: Set `HANA` in the example. It makes more sense as this formula is about HANA

File: saphanabootstrap-formula.changes [hunk 2]
```diff
@@ -1,3 +1,9 @@
+-------------------------------------------------------------------
+Tue Jan 21 14:41:29 UTC 2020 - Dario Maiocchi <dmaiocchi@suse.com>
+
+-  Version 0.4.3
+  * Add saptune module for apply a specified by user solution to each node
```
Comment (arbulu89) [inline saphanabootstrap-formula.changes]: Better `Add saptune module to apply a specific solution`. What do you think?

**Comment by MalloZup** [comment]
@arbulu89 regarding the suggested change, imho looked simply my Implementation (each node has independeance parameter) rather then having a global mechanism for all nodes and overwritten mechanism if a node want to be different. 

But i'm ok to do also the other way  thx for review !

File: pillar.example (hunk 1 above)
Comment (MalloZup) [reply pillar.example]: done thc

**Comment by MalloZup** [comment]
@arbulu89 all changes applied as requested.

I have tested the changes manually and was fine. :robot:

File: pillar.example [hunk 3]
```diff
@@ -72,7 +76,7 @@ hana:
         user: 'SYSTEM'
         password: 'Qwerty1234'
 
-    - host: hana02
+    - host: hana03
```
Comment (arbulu89) [inline pillar.example]: Don't change this. This is to show that you can have more than 1 SAP instance in the same host (this is the cost optimized scenario in fact)

File: hana/saptune.sls [hunk 4]
```diff
@@ -0,0 +1,12 @@
+{%- from "hana/map.jinja" import hana with context -%}
+{% set host = grains['host'] %}
+
+{% for node in hana.nodes if node.host == host and hana.saptune_solution is defined %}
```
Comment (arbulu89) [inline hana/saptune.sls]: I don't know if I would check `hana.saptune_solution is defined`.
And if it's not, but we have the solution in the node. This wouldn't work.
In any case we would need: 
```
{% for node in hana.nodes if node.host == host and (hana.saptune_solution is defined or node.saptune_solution is defined) %}
```

File: hana/saptune.sls [hunk 5]
```diff
@@ -0,0 +1,11 @@
+{%- from "hana/map.jinja" import hana with context -%}
+{% set host = grains['host'] %}
+
+{% for node in hana.nodes if node.host == host and (hana.saptune_solution is defined or node.saptune_solution is defined) %}
+{% set saptune_solution = node.saptune_solution|default(hana.saptune_solution) %}
+
+apply_saptune_solution_{{node.host}}{{loop.index}}:
```
Comment (arbulu89) [inline hana/saptune.sls]: I think is better to have something like:
```
{% set instance = '{:0>2}'.format(node.instance) %}
{% set name = '{}_{}'.format(node.sid, instance) %}
apply_saptune_solution_{{ host }}_{{ name }}:
```

File: pillar.example [hunk 6]
```diff
@@ -3,6 +3,10 @@ hana:
   # If set to false, this packages must be installed before the formula
   # execution manually
   # install_packages: true
+
+  # saptune solution to apply to all nodes ( by default nothing is applied)
```
Comment (arbulu89) [inline pillar.example]: Add the example of how to use `saptune_solution` in a node entry. And besides that, write a warning message that only a unique saptune solution can exist in a node, and if they are different, the formula will fail

Comment (MalloZup) [reply pillar.example]: ok done

**Pull request merged**